lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -712,7 +729,8 @@ SILCombiner::createApplyWithConcreteType(FullApplySite Apply,
   }
   // The apply can only be rewritten in terms of the concrete value if it is
   // legal to pass that value as the self argument.
-  if (CEI.isCopied && !canReplaceCopiedSelf(Apply, CEI.InitExistential, DA))
+  if (CEI.isCopied && (!CEI.InitExistential ||
+                       !canReplaceCopiedSelf(Apply, CEI.InitExistential, DA)))
```

Comment:
OK

Reply:
thinking about this a bit more: this still does not look correct to me. When types are determined from protocolconformances, even though CEI.InitExistential is null,I still want to be able to create a new Apply instruction for it. there are checks inside canReplaceCopiedSelf that I want to check as well. Does it look ok for you to add a check for returning true when InitExistential is null in canReplaceCopiedSelf (after the first check?). This way we can retain the old check.

Reply:
`canReplaceCopiedSelf` currently assumes that `self` is coming from an `init_existential`, so you should not call it in your case, at least without somehow generalizing it.

In the subsequent code, `self` is being replaced by the concrete value from the existential (`CEI.ConcreteValue`). If the existential itself was copied, then that's not generally valid!

In your case, how to you know it's safe to replace `self` with `CEI.ConcreteValue`? What if the original `self` was copied?